Modules/_ctypes/callproc.c

```diff
@@ -2022,17 +2026,15 @@ create_pointer_type(PyObject *module, PyObject *cls)
                                        "_type_", cls);
         if (result == NULL)
             return result;
-        key = Py_NewRef(cls);
     } else {
         PyErr_SetString(PyExc_TypeError, "must be a ctypes type");
         return NULL;
     }
-    if (PyDict_SetItem(st->_ctypes_ptrtype_cache, key, result) < 0) {
-        Py_DECREF(result);
-        Py_DECREF(key);
-        return NULL;
+
+    if (info) {
+        info->pointer_type = Py_XNewRef(result);
```

Comment:
```suggestion
        info->pointer_type = Py_NewRef(result);
```
`info->pointer_type` is NULL. `result` is non-NULL.

Reply:
Thanks!